functional-tests/core/api/shutdown_api_v2_test.go

```diff
@@ -30,6 +31,8 @@ var _ = Describe("/api/v2/shutdown", func() {
 			response := functional_tests.DoRequest(sling.New().Delete("http://localhost:" + hoverfly.GetAdminPort() + "/api/v2/shutdown"))
 			Expect(response.StatusCode).To(Equal(http.StatusOK))
 
+			time.Sleep(10 * time.Second)
```

Comment:
This test case is not related to the PR, but I think the test case has a bug.

What the test case expected to do is: first, it call shutdown API to kill the hoverfly instance, and then do a GET request to confirm that hoverfly is definitely closed.
- If hoverfly is closed before starting the confirm request, it receive an error "connection refused" because the remote hoverfly server is already closed. In this case, it will pass the test.
- If hoverfly is closed after starting the confirm request, it receive an error "connection reset by peer" because the remote hoverfly is closed when the confirm request is reading/writing data into socket. In this case, it fails.

I run all test cases locally many times, it just pass/fails randomly. The description says that it "should shutdown after 10 seconds", but actually there is no such source code to make it wait 10 seconds.

Reply:
I remove these changes because it is not related to this PR.

Reply:
Thank you.
Would you like to create an issue for this problem when this PR finished?

Reply:
Of course, no problem.